Flabbergasted-Flamingos/Bangazon-admin-app
# Kf computer views

# Description
Computer views added for viewing list, detail, add, and delete. One test added.
## Number of Fixes
Type of number of fixes included in this PR.

## Related Ticket(s)
#8 

## Problem to Solve
Adds access to computer table

## Proposed Changes
Computers can now be viewed, added, and deleted if no employees are assigned.

## Expected Behavior
Computers should display in list form labelled with a number. In detail view, all computer details should be displayed as well as the names of any employees using them. Clicking the add button should take you to a form page for adding a new computer which should return you to the list view when saved. A computer should not be able to be deleted if it has employees assigned to it.

## Steps to Test Solution

Run the following code
```
git fetch --all
git checkout FKComputerViews
rm -r admin_app/migrations
rm -r db.sqlite3
python manage.py makemigrations admin_app
python manage.py migrate
python manage.py seeder
```
1. Open the database in your browser
1. Assign a computer to an employee
1. Click on the computers button in the nav bar, you should see a list of computers
1. Click on a computer to view details, there should be a delete button
1. If the computer has employees assigned to it, there should be a message informing you that you cannot delete
1. If the computer has no employees assigned to it, there should be a deletion confirmation button
1. Clicking the add button should take you to a form to add a new computer
1. Saving the new computer should take you to the computers list


## Testing
Testing is not complete, code being merged at request of team lead.
[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
Documentation not complete, code being merged at request of team lead.
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
Computer model changed to fit ticket criteria.
